### ISSUE TITLE
Support clear-sign for EIP712 on Ledger

### ISSUE THREAD

**Issue by 0x398** [RabbyHub/Rabby#1816]
Ledger show only hashes when user prompted to sign EIP712 with Rabby and there is no easy way to verify what these hashes correspond to. Signing hashes is dangerous, because it might be phishy permit to transfer WETH, WBTC or monkey without expiration. Ledger supports displaying EIP712 domain and data on the device screen when signing with Ledger Live. Rabby should too! 
Might be related: https://pypi.org/project/eip712-clearsign/

**Comment by vvvvvv1vvvvvv** [comment]
Why not verify it via Rabby UI?

**Comment by 0x398** [comment]
Point of using hardware wallet is no need to trust Rabby UI (or any other software wallet). Why use hardware wallet at all if I will trust software wallet?

**Comment by vvvvvv1vvvvvv** [comment]
Got your point, does Ledger hardware support this?
As you said Ledger supports displaying EIP712 domain and data when signing with Ledger Live, it displays on Ledger Live right? So Ledger Live is software wallet in this case

**Comment by 0x398** [comment]
No, I mean when I use Ledger Live it displays full EIP712 data on the device. So, I do not need to trust Ledger Live in this case, only to Ledger device.

**Comment by vvvvvv1vvvvvv** [comment]
OK, will check how Ledger Live do this

**Comment by vvvvvv1vvvvvv** [comment]
btw, you need to click the button on device several times in this way right?
compare with hex hash only one time, don't you think it's too much?

**Comment by 0x398** [comment]
@vvvvvv1vvvvvv I edited previous answer

**Comment by 0x398** [comment]
> compare with hex hash only one time, don't you think it's too much?

No, I do not think this is too much. But if somebody thinks so, they can just disable it in the device settings and trust theirs software wallet.

**Comment by 0x398** [comment]
Maybe it is even disabled by default, @vvvvvv1vvvvvv check ETH application settings when you will look into this

**Comment by vvvvvv1vvvvvv** [comment]
will revert in #2026 since we found this will cause some issues with EIP-712 signature, will also report to Ledger team to fix them

**Comment by 0x398** [comment]
Could you rather add switch to ask user whether they want to use clear sign or blind sign? Compromising on security to workaround some rather rare bug (I personally had 0 issues with chest sign stove it was engaged) seems like a very bad decision to me. Adding switch so user can use clear sign but disable it if there are problems seems to be a lot better solution.

**Comment by vvvvvv1vvvvvv** [comment]
Have to temporary revert since other user meet the issue, no worry, will make it back after Ledger fix the issue, for track:
https://github.com/LedgerHQ/ledger-live/issues/6065

**Comment by vvvvvv1vvvvvv** [comment]
You can use specify version until we make it back, download here: https://github.com/RabbyHub/Rabby/releases/tag/v0.92.49
will update in this issue if Ledger fix it

**Comment by fafrd** [comment]
@vvvvvv1vvvvvv @heisenberg-2077 This feature was added in https://github.com/RabbyHub/Rabby/pull/2545, then removed in https://github.com/RabbyHub/Rabby/pull/2661.

Why was this removed?

During the [Radiant Capital hack](https://www.halborn.com/blog/post/explained-the-radiant-capital-hack-october-2024), multisig signers were using hardware wallets and _still_ got hacked. It's absolutely critical that clear signing is supported -- organizations cannot rely on Rabby alone when their machines could be compromised.
